rust/feature-flags/tests/test_flags.rs

```diff
@@ -3476,3 +3476,170 @@ async fn test_disable_flags_without_config_param_has_minimal_response() -> Resul
 
     Ok(())
 }
+
+#[tokio::test]
+async fn test_numeric_group_ids_work_correctly() -> Result<()> {
+    let config = DEFAULT_TEST_CONFIG.clone();
+    let distinct_id = "user_with_numeric_group".to_string();
+    let redis_client = setup_redis_client(Some(config.redis_url.clone()));
+    let pg_client = setup_pg_reader_client(None).await;
+    let team_id = rand::thread_rng().gen_range(1..10_000_000);
+    let team = insert_new_team_in_pg(pg_client.clone(), Some(team_id))
+        .await
+        .unwrap();
+
+    insert_person_for_team_in_pg(pg_client.clone(), team.id, distinct_id.clone(), None).await?;
+
+    let token = team.api_token;
+
+    // Create a group with a numeric group_key (as a string in DB, but represents a number)
+    create_group_in_pg(
+        pg_client.clone(),
+        team.id,
+        "organization",
+        "123",
+        json!({"name": "Organization 123", "size": "large"}),
+    )
+    .await?;
+
+    // Define a group-based flag with simple rollout (no property filters)
+    let flags_json = json!([
+        {
+            "id": 1,
+            "key": "numeric-group-flag",
+            "name": "Flag targeting numeric group ID",
+            "active": true,
+            "deleted": false,
+            "team_id": team.id,
+            "filters": {
+                "aggregation_group_type_index": 1,
+                "groups": [
+                    {
+                        "properties": [],
+                        "rollout_percentage": 100
+                    }
+                ]
+            }
+        }
+    ]);
+
+    insert_flags_for_team_in_redis(
+        redis_client.clone(),
+        team.id,
+        team.project_id,
+        Some(flags_json.to_string()),
+    )
+    .await?;
+
+    let server = ServerHandle::for_config(config).await;
+
+    // Test with numeric group ID (integer in JSON, not string)
+    {
+        let payload = json!({
+            "token": token,
+            "distinct_id": distinct_id,
+            "groups": {
+                "organization": 123  // This is a JSON number, not a string
+            }
+        });
+
+        let res = server
+            .send_flags_request(payload.to_string(), None, None)
+            .await;
+        assert_eq!(res.status(), StatusCode::OK);
+
+        let json_data = res.json::<Value>().await?;
+        assert_json_include!(
+            actual: json_data,
+            expected: json!({
+                "errorsWhileComputingFlags": false,
+                "featureFlags": {
+                    "numeric-group-flag": true
+                }
+            })
+        );
+    }
```

Comment:
I was confused by the curly braces here but I learned it creates a new scope so you can re-use the same variables for each of these test cases. Neat!

Reply:
yeah fun short hand